package.json

```diff
@@ -49,7 +49,7 @@
     "grunt-legacy-util": "~1.1.1",
     "iconv-lite": "~0.4.13",
     "js-yaml": "~3.5.2",
-    "minimatch": "~3.0.2",
+    "minimatch": "^3.0.4",
```

Comment:
please set `~` here

Reply:
@vladikoff I can do that, no problem. I'm wondering though, if you could confirm the behaviour I'm experiencing. Which is that `~3.0.2` means npm always installs and uses 3.0.2 for grunt unless the project or other dependency specifies a newer, compatible, version. Whereas with `^3.0.2` I found it consistently prefers newer versions where possible.

Reply:
should be same

![image](https://user-images.githubusercontent.com/128755/44418700-1518ad80-a547-11e8-95e6-c4694d885a96.png)

we use `~` for consistance with the rest of the dependencies and to avoid breakage

Reply:
@Krinkle Check your `package-lock.json`. The current range `~3.0.2` installs `minimatch@3.0.4` on fresh installs.

Reply:
@vladikoff I understand that the tilde operator allows expansion to minor versions.

But there could be a difference between compatibility allowance (for de-duplication), and default preference for expansion (during installation when there are no duplicates from other packages).

When I added `"grunt": "~1.0.3"` to a project yesterday, and ran `rm -rf node_modules; npm install`, I received `minimatch@3.0.2`.